internal/webhook/template_webhook.go

```diff
@@ -55,7 +64,31 @@ func (*ClusterTemplateValidator) ValidateUpdate(_ context.Context, _ runtime.Obj
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ClusterTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (v *ClusterTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	template, ok := obj.(*v1alpha1.ClusterTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ClusterTemplate but got a %T", obj))
+	}
+
+	managedClusters := &v1alpha1.ManagedClusterList{}
+	listOptions := client.ListOptions{
```

Comment:
Currently, `ManagedCluster` may refer the `ClusterTemplate` from the system namespace only. It will be changed soon but until then, let's validate the removal of the `ClusterTemplates` in the system namespace only.

Reply:
I can, but previously it was requested to search across all namespaces https://github.com/Mirantis/hmc/pull/301#discussion_r1760023794

Reply:
Sorry, I probably wasn't clear enough. These suggestions are a little bit different. In https://github.com/Mirantis/hmc/pull/301#discussion_r1760023794 I meant to list `ManagedClusters` across all namespaces while trying to find clusters that refer to the `ClusterTemplate` planned to be removed. But in this comment, I'm suggesting validating the removal of `ClusterTemplate` objects in the system namespace only. Since now, all templates that are not in the system namespace can't be referenced and thus, can be removed without any extra validation. But it's not that important since this logic will be changed soon (after #317  `ManagedCluster` will be allowed to refer to the `ClusterTemplate` from the same namespace only), so, you can ignore it.